### PR TITLE
[NETBEANS-4681] Fixed issue : refactor record rename & move record fi…

### PR DESCRIPTION
**Pull request by singh-akhilesh**
…le to other package

https://issues.apache.org/jira/browse/NETBEANS-4681

### CODE REVIEW

**Comment by singh-akhilesh** [comment]
@junichi11 Please review and approved your comments.

File: java/refactoring.java/src/org/netbeans/modules/refactoring/java/ui/MoveClassUI.java [hunk 1]
```diff
@@ -305,7 +305,11 @@ public RefactoringUI create(CompilationInfo info, TreePathHandle[] handles, File
                 Collection<TreePathHandle> tphs = new ArrayList<TreePathHandle>();
                 SourcePositions sourcePositions = info.getTrees().getSourcePositions();
                 for (Element ele : e.getEnclosedElements()) {
-                    Tree leaf = info.getTrees().getPath(ele).getLeaf();
+                    TreePath path = info.getTrees().getPath(ele);
+                    if(path == null){
+                        continue;
+                    }
```
Comment (junichi11) [inline java/refactoring.java/src/org/netbeans/modules/refactoring/java/ui/MoveClassUI.java:311]: Please reformat.

**Comment by junichi11** [comment]
As I wrote before, I'm not an expert of this area. So, I've just checked code style.

**Comment by junichi11** [comment]
@jlahoda Could you please review this if you can make time? Thanks.

File: java/refactoring.java/src/org/netbeans/modules/refactoring/java/ui/MoveClassUI.java (hunk 1 above)
Comment (singh-akhilesh) [reply java/refactoring.java/src/org/netbeans/modules/refactoring/java/ui/MoveClassUI.java]: comment addressed

**Comment by lahodaj** [comment]
Looks good to me.

**Comment by singh-akhilesh** [comment]
@lahodaj Please approve the PR.

**Review by jlahoda — APPROVED** [review]

**Comment by arvindaprameya** [comment]
Approved,

File: java/refactoring.java/src/org/netbeans/modules/refactoring/java/ui/MoveClassUI.java [hunk 2]
```diff
@@ -305,7 +305,11 @@ public RefactoringUI create(CompilationInfo info, TreePathHandle[] handles, File
                 Collection<TreePathHandle> tphs = new ArrayList<TreePathHandle>();
                 SourcePositions sourcePositions = info.getTrees().getSourcePositions();
                 for (Element ele : e.getEnclosedElements()) {
-                    Tree leaf = info.getTrees().getPath(ele).getLeaf();
+                    TreePath path = info.getTrees().getPath(ele);
+                    if (path == null) {
```
Comment (lahodaj) [inline java/refactoring.java/src/org/netbeans/modules/refactoring/java/ui/MoveClassUI.java:309]: I assume this is null for the synthetic record members, etc., right?

File: java/java.source.base/src/org/netbeans/api/java/source/TreeUtilities.java [hunk 3]
```diff
@@ -124,7 +125,14 @@
      * @since 0.67
      */
     public static final Set<Kind> CLASS_TREE_KINDS = EnumSet.of(Kind.ANNOTATION_TYPE, Kind.CLASS, Kind.ENUM, Kind.INTERFACE);
-    
+    static {
```
Comment (lahodaj) [inline java/java.source.base/src/org/netbeans/api/java/source/TreeUtilities.java:128]: We could make CLASS_TREE_KINDS immutable, although not necessarily in this PR.

**Review by lahodaj — APPROVED** [review]
Looks reasonable to me.

**Comment by arvindaprameya** [comment]
Approved, Thanks for completing this

File: java/refactoring.java/src/org/netbeans/modules/refactoring/java/ui/MoveClassUI.java (hunk 2 above)
Comment (singh-akhilesh) [reply java/refactoring.java/src/org/netbeans/modules/refactoring/java/ui/MoveClassUI.java:309]: Yes, Its for synthetic record members.

File: java/java.source.base/src/org/netbeans/api/java/source/TreeUtilities.java (hunk 3 above)
Comment (singh-akhilesh) [reply java/java.source.base/src/org/netbeans/api/java/source/TreeUtilities.java:128]: Yes, we could. Will make immutable later.

**Pull request merged**